### PR TITLE
Implement end of stream detection in the HTML5 provider 

### PR DESCRIPTION
**Pull request by johnBartos**
JW7-3269

### CODE REVIEW

File: src/js/providers/html5.js [hunk 1]
```diff
@@ -998,6 +1015,48 @@ define([
...
+
+            if (live && _lastEndOfBuffer === endOfBuffer) {
+                if (!_staleStreamTimeout) {
+                    _staleStreamTimeout = setTimeout(function () {
+                        _stale = true;
+                        checkStreamEnded();
+                    }, _staleStreamDuration);
+                }
+            } else {
+                clearTimeout(_staleStreamTimeout);
```
Comment (robwalch) [inline src/js/providers/html5.js]: This should probably be called anywhere we call `clearTimeout(_playbackTimeout);`

File: src/js/providers/html5.js [hunk 2]
```diff
@@ -998,6 +1015,48 @@ define([
...
+                });
+                return true;
+            }
+
+            return false;
+        }
+
+        function atEndOfStream() {
+            if (_videotag.duration !== Infinity) {
+                return;
```
Comment (robwalch) [inline src/js/providers/html5.js]: return false;

File: src/js/providers/html5.js [hunk 3]
```diff
@@ -998,6 +1015,48 @@ define([
...
+                _this.trigger(events.JWPLAYER_MEDIA_ERROR, {
+                    message: 'The live stream is either down or has ended'
+                });
+                return true;
+            }
+
+            return false;
+        }
+
+        function atEndOfStream() {
```
Comment (robwalch) [inline src/js/providers/html5.js]: How about `atEdgeOfLiveStream`?

File: src/js/providers/html5.js (hunk 1 above)
Comment (johnBartos) [inline src/js/providers/html5.js]: done

**Pull request merged**